tests/pg_qs_test_runner.py

```diff
@@ -28,77 +32,97 @@ class TeardownException(Exception): pass
-        test_formats,
-        test_timing_buffers_conflicts,
-        test_insert_on_conflict,
-        ]
+	test_deadlock,
+	test_simple_query,
+	test_concurrent_access,
+	test_nested_call,
+	test_trigger,
+	test_costs,
+	test_buffers,
+	test_timing,
+	test_formats,
+	test_timing_buffers_conflicts,
+	test_insert_on_conflict,
+]
 
 def setup(con):
 	''' Creates pg_query_state extension, creates tables for tests, fills it with data '''
-	print 'setting up...'
+	print('setting up...')
 	try:
 		cur = con.cursor()
 		for cmd in setup_cmd:
 			cur.execute(cmd)
 		con.commit()
 		cur.close()
-	except Exception, e:
+	except Exception as e:
 		raise SetupException('Setup failed: %s' % e)
-	print 'done!'
+	print('done!')
 
 def teardown(con):
 	''' Drops table and extension '''
-	print 'tearing down...'
+	print('tearing down...')
 	try:
 		cur = con.cursor()
 		for cmd in teardown_cmd:
 			cur.execute(cmd)
 		con.commit()
 		cur.close()
-	except Exception, e:
+	except Exception as e:
 		raise TeardownException('Teardown failed: %s' % e)
-	print 'done!'
+	print('done!')
 
 def main(config):
 	''' Main test function '''
-	con = psycopg2.connect(**config)
-	setup(con)
+	conn_params = {
+		key:config.__dict__[key] for key in ('host', 'port', 'user', 'database', 'password')
+	}
+
+	if config.tpcds_setup:
+		print('Setup database for TPC-DS bench')
+		tpcds.setup_tpcds(conn_params)
+		print('Database is setup successfully')
+		return
+
+	if config.tpcds_run:
+		print('Starting stress test')
+		tpcds.run_tpcds(conn_params)
+		print('Stress finished successfully')
+		return
 
+	# run default tests
+	init_conn = psycopg2.connect(**conn_params)
+	setup(init_conn)
 	for i, test in enumerate(tests):
 		if test.__doc__:
 			descr = test.__doc__
 		else:
 			descr = 'test case %d' % (i+1)
-		print ("%s..." % descr),; sys.stdout.flush()
-		test(config)
-		print 'ok!'
-
-	teardown(con)
-	con.close()
+		print(("%s..." % descr))
+		sys.stdout.flush()
+		test(conn_params)
+		print('ok!')
+	teardown(init_conn)
+	init_conn.close()
 
 if __name__ == '__main__':
 	parser = argparse.ArgumentParser(description='Query state of running backends tests')
+
 	parser.add_argument('--host', default='localhost', help='postgres server host')
 	parser.add_argument('--port', type=int, default=5432, help='postgres server port')
 	parser.add_argument('--user', dest='user', default='postgres', help='user name')
 	parser.add_argument('--database', dest='database', default='postgres', help='database name')
-	parser.add_argument('--password', dest='password', nargs=0, action=PasswordPromptAction, default='')
+	parser.add_argument('--password', dest='password', nargs=0, action=PasswordPromptAction, default='', help='password')
+	parser.add_argument('--tpc-ds-setup', dest='tpcds_setup', action='store_true', help='setup database to run TPC-DS benchmark')
+	parser.add_argument('--tpc-ds-run', dest='tpcds_run', action='store_true', help='run only stress test based on TPC-DS benchmark')
```

Comment:
Because the whole task(TPC-DS running) is huge and its running takes much time. Clearly, it have to be decomposed to separate subtasks so to be able to run their more granular. This separation is the fist step to achieve it. In particular, it provides to run tpc-ds bench multiple times for debugging purpose without initializing database at all time.
Yes, `tpc-ds-run` requires `tpc-ds-setup` and fails until database is initialized properly.